src/main/java/kitchenpos/application/TableService.java

```diff
@@ -22,53 +27,46 @@ public TableService(final OrderDao orderDao, final OrderTableDao orderTableDao)
     }
 
     @Transactional
-    public OrderTable create(final OrderTable orderTable) {
-        orderTable.setId(null);
-        orderTable.setTableGroupId(null);
+    public OrderTableResponse create(final OrderTableCreateRequest request) {
+        final OrderTable orderTable = orderTableDao.save(request.toEntity());
 
-        return orderTableDao.save(orderTable);
+        return OrderTableResponse.from(orderTable);
     }
 
-    public List<OrderTable> list() {
-        return orderTableDao.findAll();
+    public List<OrderTableResponse> list() {
+        return orderTableDao.findAll()
+            .stream()
+            .map(OrderTableResponse::from)
+            .collect(Collectors.toList());
     }
 
     @Transactional
-    public OrderTable changeEmpty(final Long orderTableId, final OrderTable orderTable) {
-        final OrderTable savedOrderTable = orderTableDao.findById(orderTableId)
-                .orElseThrow(IllegalArgumentException::new);
-
-        if (Objects.nonNull(savedOrderTable.getTableGroupId())) {
-            throw new IllegalArgumentException();
-        }
-
-        if (orderDao.existsByOrderTableIdAndOrderStatusIn(
-                orderTableId, Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))) {
-            throw new IllegalArgumentException();
-        }
-
-        savedOrderTable.setEmpty(orderTable.isEmpty());
-
-        return orderTableDao.save(savedOrderTable);
+    public OrderTableResponse changeEmpty(final Long orderTableId,
+        final OrderTableUpdateEmptyRequest request
+    ) {
+        final OrderTable orderTable = orderTableDao.findById(orderTableId)
+            .orElseThrow(IllegalArgumentException::new);
+        validateCompletion(orderTableId);
```

Comment:
아래처럼 올바르지 않은 테이블 주문 상태인 경우 아예 예외처리를 먼저 해버리는 순서로 하는 것은 어떤가요?
```suggestion
        validateCompletion(orderTableId);
        final OrderTable orderTable = orderTableDao.findById(orderTableId)
            .orElseThrow(IllegalArgumentException::new);
```